Decred Wallet/Features/Transactions/TransactionsViewController.swift

```diff
@@ -67,121 +70,115 @@ class TransactionHistoryViewController: UIViewController {
 
         self.transactionFilterDropDown.initMenu(filterOptions) { [weak self] index, value in
-            self?.applyTxFilter(currentFilter: self!.transactionFilters[index])
+             self?.reloadTxsForCurrentFilter()
         }
-        self.transactionFilterDropDown.setSelectedItemIndex(0)
     }
-    
+
     @objc func reloadTxsForCurrentFilter() {
-        var currentFilterItem = DcrlibwalletTxFilterAll
-        if self.transactionFilterDropDown.selectedItemIndex >= 0 && self.transactionFilters.count > self.transactionFilterDropDown.selectedItemIndex {
-            currentFilterItem = self.transactionFilters[self.transactionFilterDropDown.selectedItemIndex]
-        }
-        self.applyTxFilter(currentFilter: currentFilterItem)
-    }
-    
-    func applyTxFilter(currentFilter: Int32) {
+        self.allTransactions.removeAll()
         self.refreshControl.showLoader(in: self.transactionsTableView)
-        
+
         defer {
             self.transactionsTableView.reloadData()
             self.refreshControl.endRefreshing()
         }
-        
-        switch currentFilter {
-        case DcrlibwalletTxFilterSent:
-            self.filteredTransactions = self.allTransactions.filter{$0.direction == DcrlibwalletTxDirectionSent && $0.type == DcrlibwalletTxTypeRegular}
-            break
-            
-        case DcrlibwalletTxFilterReceived:
-            self.filteredTransactions = self.allTransactions.filter{$0.direction == DcrlibwalletTxDirectionReceived && $0.type == DcrlibwalletTxTypeRegular}
-            break
-            
-        case DcrlibwalletTxFilterTransferred:
-            self.filteredTransactions = self.allTransactions.filter{$0.direction == DcrlibwalletTxDirectionTransferred && $0.type == DcrlibwalletTxTypeRegular}
-            break
-            
-        case DcrlibwalletTxFilterStaking:
-            self.filteredTransactions = self.allTransactions.filter{$0.type == DcrlibwalletTxTypeRevocation || $0.type == DcrlibwalletTxTypeTicketPurchase || $0.type == DcrlibwalletTxTypeVote }
-            break
-            
-        case DcrlibwalletTxFilterCoinBase:
-            self.filteredTransactions = self.allTransactions.filter{$0.type == DcrlibwalletTxTypeCoinBase}
-            break
-            
-        default:
-            self.filteredTransactions.removeAll()
-            break
+
+        var currentFilterItem = DcrlibwalletTxFilterAll
+        if self.transactionFilterDropDown.selectedItemIndex >= 0 && self.transactionFilters.count > self.transactionFilterDropDown.selectedItemIndex {
+            currentFilterItem = self.transactionFilters[self.transactionFilterDropDown.selectedItemIndex]
+        }
+
+        var currentSorterType = true
+        if self.transactionSorterDropDown.selectedItemIndex >= 0 && self.transactionSorters.count > self.transactionSorterDropDown.selectedItemIndex {
+            currentSorterType = self.transactionSorters[self.transactionSorterDropDown.selectedItemIndex]
+        }
```

Comment:
These can be simplified thus:
```
let selectedFilterIndex = self.transactionFilterDropDown.selectedItemIndex
let currentFilterItem = self.transactionFilters[safe: selectedFilterIndex] ?? DcrlibwalletTxFilterAll

let selectedSortIndex = self.transactionSorterDropDown.selectedItemIndex
let sortOrderNewerFirst = self.transactionSorters[safe: selectedSortIndex] ?? true
```

You'd need to add the following extension method to `Extensions.swift`:
```
extension Collection {
    // Returns the element at the specified index if it is within bounds, otherwise nil.
    subscript (safe index: Index) -> Element? {
        return indices.contains(index) ? self[index] : nil
    }
}
```